operator/pkg/utils/patch/patch.go

```diff
@@ -23,6 +23,47 @@ import (
 	"k8s.io/apimachinery/pkg/util/strategicpatch"
 )
 
+// VolumeClaimTemplateSpec returns the merged VolumeClaimTemplate spes
+func VolumeClaimTemplateSpec(defaultSpec, patch []v1.PersistentVolumeClaim) ([]v1.PersistentVolumeClaim, error) {
+	if len(patch) == 0 {
+		return defaultSpec, nil
+	}
+	defaults, patches := map[string]v1.PersistentVolumeClaim{}, map[string]v1.PersistentVolumeClaim{}
+	for _, spec := range defaultSpec {
+		if _, ok := defaults[spec.Name]; ok {
+			return nil, fmt.Errorf("duplicate PersistentVolumeClaim detected, Name: %s", spec.ObjectMeta.Name)
```

Comment:
Now that I see this, maybe hard to avoid erroring.

Reply:
Yeah, when we need to process arbitrary user input, things tense to go wrong by a lot. Can't really avoid it.